### PR TITLE
Color Settings test cases

### PR DESCRIPTION
**Pull request by lukewalczak**
Test cases for complete color settings mechanism

Merge after: https://github.com/WordPress/gutenberg/pull/22493

### CODE REVIEW

File: test-cases/gutenberg/color-settings.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+
+# Color Settings - Test Cases
+
+#### **Precondition**
+
+Added block with color support such as `Buttons` or `Cover`
```
Comment (pinarol) [inline test-cases/gutenberg/color-settings.md:6]: I think we should add the steps to open the color settings and remove "Press `Background Color` / `Color`" from the rest

File: test-cases/gutenberg/color-settings.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+**Steps:**
+
+* Press `Background Color` / `Color`
+* Switch several times between `Solid` and `Gradient` segments 
+
+**Expected behavior:**
+
+* Selection animation works smoothly
+* Color swatches fade parallel with active segment change
```
Comment (pinarol) [inline test-cases/gutenberg/color-settings.md:22]: it is not very obvious what smoothly means, could you add a gif to decribe?

File: test-cases/gutenberg/color-settings.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+**Steps:**
+
+* Press `Background Color` / `Color`
+* Select one of the last solid colors (_blue or purple_)
+* Select `Gradient` segment
+
+**Expected behavior:**
+
+* Color list is scrolled to the start
+* Color swatches fade parallel with scroll
```
Comment (pinarol) [inline test-cases/gutenberg/color-settings.md:39]: Again a gif would really help, I don't know what to expect by just reading

File: test-cases/gutenberg/color-settings.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+**Steps:**
+
+* Press `Background Color` / `Color`
+* Press any color several times
+
+**Expected behavior:**
+
+* Color swatch is bouncing on every press
+* Selected icon on white background fades in over the color swatch (_only on the first press_)
```
Comment (pinarol) [inline test-cases/gutenberg/color-settings.md:73]: Same: a gif would be good

File: test-cases/gutenberg/color-settings.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+* Colors list is always scrolled to the end (to the _custom_ color) when custom color is selected
+
+**Next Steps:**
+
+* Select one of the last gradient colors available on the screen
+
+**Expected behavior:**
+
+* List is scrolled and `Custom` button slides out
+
```
Comment (pinarol) [inline test-cases/gutenberg/color-settings.md:132]: Same: a gif would be good

File: test-cases/gutenberg/color-settings.md (hunk 2 above)
Comment (lukewalczak) [reply test-cases/gutenberg/color-settings.md:22]: done!

File: test-cases/gutenberg/color-settings.md (hunk 3 above)
Comment (lukewalczak) [reply test-cases/gutenberg/color-settings.md:40]: done!

File: test-cases/gutenberg/color-settings.md (hunk 4 above)
Comment (lukewalczak) [reply test-cases/gutenberg/color-settings.md:76]: done!

File: test-cases/gutenberg/color-settings.md (hunk 5 above)
Comment (lukewalczak) [reply test-cases/gutenberg/color-settings.md:143]: done!

**Pull request merged**